### ISSUE TITLE
[v1.0.1] `options.parallel` on windows doesn't speed up compression (windows)

### ISSUE THREAD

**Issue by mcongy** [webpack-contrib/uglifyjs-webpack-plugin#148]
Hello,

Parallel option doesn't seem to work on windows.

uglifyjs-webpack-plugin version: 1.0.1
webpack: 3.8.1
Windows: 10 (Version 1703, build 15063.674)

Tried using parallel option (both as a top-level argument to `UglifyJSPlugin({ parallel: true }) ` and as a `uglifyOptions: { parallel: true }`
Tried both using `true` and specific number of threads.

Tried doing 50+ tests, and I didn't see any performance improvement.

**Comment by evilebottnawi** [comment]
@mcongy please provide information about you node and computer information (count core CPU, memory and etc)

**Comment by mcongy** [comment]
Node: 8.7.0 (default installation using https://chocolatey.org/ )
CPU: Core i9 7900X (10 cores, 20 threads), 3.3Ghz
RAM: 32GB
64bit Win 10 Pro

**Comment by evilebottnawi** [comment]
@mcongy good, I will wait

**Comment by michael-ciniawsky** [comment]
@mcongy Please include your `webpack.config.js` aswell, how many entries does your application have? Maybe related to #118 

How many entries does your application have?
Could you check all running processes via the Task Manager while running `webpack` (#118)
Do you use caching `{ cache: true } or just `parallel` ?

> Caching brought the most benefits in my case, parallel aswell but not _that_ much tbh

**Comment by mcongy** [comment]
Hello, I tried it on other Windows PC.

Node: 8.8.0
Windows 10 Education N
Core i3 7100 (2 cores, 4 threads, 3.7 GHz)
16 GB RAM

Uglify config:

```    
new UglifyJSPlugin({
  parallel: false,
  uglifyOptions: {
    parallel: true,
    sourceMap: process.env.APP_ENV === 'production',
    output: {
      comments: false
    }
  }
})
```
App have only 1 entrypoint

Results:
No performance improvement. (All test did vary from 15.4 to 15.6 seconds no matter the parallel option).

**Comment by michael-ciniawsky** [comment]
How many assets (bundles) do you get ?  2 Cores === 1 Possibility to process of the main 'thread'.
When you run webpack could you inspect how many `node.exe` processes are spawned? (Task Manager)

**Comment by michael-ciniawsky** [comment]
Besides that fact and after some experimentation, please use `{ cache: true }` 😛

**Comment by d3viant0ne** [comment]
The vast majority of the performance improvements are going to come from caching not parallel execution though you will see performance improvements in cache generation with parallel enabled.

**Comment by mcongy** [comment]
Ok, thanks for the explanation. Consider adding it to the docs, so it won't confuse other people.

I have one more question. Does `cache` options has some downfalls (e.g. having to manually invalidate the cache in certain cases)?

Thanks!

**Comment by d3viant0ne** [comment]
Oddly enough, I had that should be in the docs discussion a few days ago :/

I've done a fair bit of testing with the cache enabled and I haven't seen any issues that would make me stop using it professionally. I also pulled some basic execution time numbers right before it was released that can be found [here](https://twitter.com/WebpackContrib/status/922276019700224001).

I pulled flame graphs as well but given you can't really compare 0.4.6 with 1.0 the only real comparison would have been cache on / off & parallel on / off. They were similar enough to not be of much value.

The cache handles itself, no manual intervention needed. If you want to inspect it you can find it in `./node_modules/.cache`.

**Comment by d3viant0ne** [comment]
I'll see if I can find some time this weekend to do a bit more testing around the parallel option without the cache. There may very well be an issue preventing the plugin to not leverage the parallelization effectively, my focus for the initial testing was focused on the caching and the possibility of swapping out the library that manages it.

For reference, it uses `cacache` https://github.com/zkat/cacache which has outperformed any other on disk cache that could have been a viable replacement.

**Comment by michael-ciniawsky** [comment]
Fixed in #173 && released in `v1.1.0`
